src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts

```diff
@@ -0,0 +1,114 @@
+import { createEvent, createStore, forward, sample } from 'effector';
+import { combineEvents } from 'patronum';
+
+import { walletConnectModel, type InitReconnectParams } from '@renderer/entities/walletConnect';
+import { toAccountId } from '@renderer/shared/lib/utils';
+import { chainsService } from '@renderer/entities/network';
+import { WalletConnectAccount } from '@renderer/shared/core';
+import { ReconnectStep } from '../lib/constants';
+import { walletModel } from '@renderer/entities/wallet';
+import { isConnectedStep, isReconnectingStep } from '../lib/utils';
+import { signModel } from './sign-model';
+
+const reset = createEvent();
+const reconnectModalShown = createEvent();
+const reconnectStarted = createEvent<InitReconnectParams>();
+const reconnectAborted = createEvent();
+const reconnectDone = createEvent();
+
+const $reconnectStep = createStore(ReconnectStep.NOT_STARTED).reset(reset);
+
+sample({
+  clock: reconnectModalShown,
+  fn: () => ReconnectStep.READY_TO_RECONNECT,
+  target: $reconnectStep,
+});
+
+sample({
+  clock: reconnectStarted,
+  fn: () => ReconnectStep.RECONNECTING,
+  target: $reconnectStep,
+});
+
+forward({
+  from: reconnectStarted,
+  to: walletConnectModel.events.connect,
+});
+
+sample({
+  clock: walletConnectModel.events.connected,
+  source: {
+    step: $reconnectStep,
+    session: walletConnectModel.$session,
+  },
+  filter: ({ step, session }) => isReconnectingStep(step) || (isConnectedStep(step) && Boolean(session?.topic)),
```

Comment:
I like this `filter`
Could we somehow add `Boolean(session?.topic)` inside utility function?